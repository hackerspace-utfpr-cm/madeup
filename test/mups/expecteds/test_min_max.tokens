1(0-1) ID: f1
1(3-3) ASSIGN: =
1(5-7) REAL: 8.6
1(8-8) NEWLINE: \n
2(9-10) ID: f2
2(12-12) ASSIGN: =
2(14-16) REAL: 9.1
2(17-17) NEWLINE: \n
3(18-18) NEWLINE: \n
4(19-20) ID: i1
4(22-22) ASSIGN: =
4(24-24) INTEGER: 3
4(25-25) NEWLINE: \n
5(26-27) ID: i2
5(29-29) ASSIGN: =
5(31-31) INTEGER: 4
5(32-32) NEWLINE: \n
6(33-33) NEWLINE: \n
7(34-38) ID: debug
7(40-40) PIPE: |
7(41-41) LEFT_PARENTHESIS: (
7(42-44) ID: min
7(46-47) ID: f1
7(48-48) COMMA: ,
7(50-51) ID: f2
7(52-52) RIGHT_PARENTHESIS: )
7(54-54) MINUS: -
7(56-57) ID: f1
7(58-58) PIPE: |
7(60-60) LESS_THAN: <
7(62-66) REAL: 0.001
7(67-67) NEWLINE: \n
8(68-72) ID: debug
8(74-74) PIPE: |
8(75-75) LEFT_PARENTHESIS: (
8(76-78) ID: max
8(80-81) ID: f1
8(82-82) COMMA: ,
8(84-85) ID: f2
8(86-86) RIGHT_PARENTHESIS: )
8(88-88) MINUS: -
8(90-91) ID: f2
8(92-92) PIPE: |
8(94-94) LESS_THAN: <
8(96-100) REAL: 0.001
8(101-101) NEWLINE: \n
9(102-106) ID: debug
9(108-108) LEFT_PARENTHESIS: (
9(109-111) ID: min
9(113-114) ID: i1
9(115-115) COMMA: ,
9(117-118) ID: i2
9(119-119) RIGHT_PARENTHESIS: )
9(121-122) EQUAL_TO: ==
9(124-125) ID: i1
9(126-126) NEWLINE: \n
10(127-131) ID: debug
10(133-133) LEFT_PARENTHESIS: (
10(134-136) ID: max
10(138-139) ID: i1
10(140-140) COMMA: ,
10(142-143) ID: i2
10(144-144) RIGHT_PARENTHESIS: )
10(146-147) EQUAL_TO: ==
10(149-150) ID: i2
10(151-151) NEWLINE: \n
11(152-152) NEWLINE: \n
12(153-157) ID: debug
12(159-159) PIPE: |
12(160-160) LEFT_PARENTHESIS: (
12(161-163) ID: min
12(165-166) ID: f1
12(167-167) COMMA: ,
12(169-170) ID: i1
12(171-171) RIGHT_PARENTHESIS: )
12(173-173) MINUS: -
12(175-176) ID: i1
12(177-177) PIPE: |
12(179-179) LESS_THAN: <
12(181-185) REAL: 0.001
12(186-186) NEWLINE: \n
13(187-191) ID: debug
13(193-193) PIPE: |
13(194-194) LEFT_PARENTHESIS: (
13(195-197) ID: min
13(199-200) ID: f1
13(201-201) COMMA: ,
13(203-204) ID: i2
13(205-205) RIGHT_PARENTHESIS: )
13(207-207) MINUS: -
13(209-210) ID: i2
13(211-211) PIPE: |
13(213-213) LESS_THAN: <
13(215-219) REAL: 0.001
13(220-220) NEWLINE: \n
14(221-225) ID: debug
14(227-227) PIPE: |
14(228-228) LEFT_PARENTHESIS: (
14(229-231) ID: max
14(233-234) ID: f1
14(235-235) COMMA: ,
14(237-238) ID: i1
14(239-239) RIGHT_PARENTHESIS: )
14(241-241) MINUS: -
14(243-244) ID: f1
14(245-245) PIPE: |
14(247-247) LESS_THAN: <
14(249-253) REAL: 0.001
14(254-254) NEWLINE: \n
15(255-259) ID: debug
15(261-261) PIPE: |
15(262-262) LEFT_PARENTHESIS: (
15(263-265) ID: max
15(267-268) ID: f1
15(269-269) COMMA: ,
15(271-272) ID: i2
15(273-273) RIGHT_PARENTHESIS: )
15(275-275) MINUS: -
15(277-278) ID: f1
15(279-279) PIPE: |
15(281-281) LESS_THAN: <
15(283-287) REAL: 0.001
15(288-288) NEWLINE: \n
16(289-289) NEWLINE: \n
17(290-294) ID: debug
17(296-296) PIPE: |
17(297-297) LEFT_PARENTHESIS: (
17(298-300) ID: min
17(302-303) ID: i1
17(304-304) COMMA: ,
17(306-307) ID: f1
17(308-308) RIGHT_PARENTHESIS: )
17(310-310) MINUS: -
17(312-313) ID: i1
17(314-314) PIPE: |
17(316-316) LESS_THAN: <
17(318-322) REAL: 0.001
17(323-323) NEWLINE: \n
18(324-328) ID: debug
18(330-330) PIPE: |
18(331-331) LEFT_PARENTHESIS: (
18(332-334) ID: min
18(336-337) ID: i2
18(338-338) COMMA: ,
18(340-341) ID: f1
18(342-342) RIGHT_PARENTHESIS: )
18(344-344) MINUS: -
18(346-347) ID: i2
18(348-348) PIPE: |
18(350-350) LESS_THAN: <
18(352-356) REAL: 0.001
18(357-357) NEWLINE: \n
19(358-362) ID: debug
19(364-364) PIPE: |
19(365-365) LEFT_PARENTHESIS: (
19(366-368) ID: max
19(370-371) ID: i1
19(372-372) COMMA: ,
19(374-375) ID: f1
19(376-376) RIGHT_PARENTHESIS: )
19(378-378) MINUS: -
19(380-381) ID: f1
19(382-382) PIPE: |
19(384-384) LESS_THAN: <
19(386-390) REAL: 0.001
19(391-391) NEWLINE: \n
20(392-396) ID: debug
20(398-398) PIPE: |
20(399-399) LEFT_PARENTHESIS: (
20(400-402) ID: max
20(404-405) ID: i2
20(406-406) COMMA: ,
20(408-409) ID: f1
20(410-410) RIGHT_PARENTHESIS: )
20(412-412) MINUS: -
20(414-415) ID: f1
20(416-416) PIPE: |
20(418-418) LESS_THAN: <
20(420-424) REAL: 0.001
20(425-425) NEWLINE: \n
21(426-426) NEWLINE: \n
22(427-431) ID: debug
22(433-433) PIPE: |
22(434-434) LEFT_PARENTHESIS: (
22(435-437) ID: min
22(439-440) ID: i1
22(441-441) COMMA: ,
22(443-444) ID: f2
22(445-445) RIGHT_PARENTHESIS: )
22(447-447) MINUS: -
22(449-450) ID: i1
22(451-451) PIPE: |
22(453-453) LESS_THAN: <
22(455-459) REAL: 0.001
22(460-460) NEWLINE: \n
23(461-465) ID: debug
23(467-467) PIPE: |
23(468-468) LEFT_PARENTHESIS: (
23(469-471) ID: min
23(473-474) ID: i2
23(475-475) COMMA: ,
23(477-478) ID: f2
23(479-479) RIGHT_PARENTHESIS: )
23(481-481) MINUS: -
23(483-484) ID: i2
23(485-485) PIPE: |
23(487-487) LESS_THAN: <
23(489-493) REAL: 0.001
23(494-494) NEWLINE: \n
24(495-499) ID: debug
24(501-501) PIPE: |
24(502-502) LEFT_PARENTHESIS: (
24(503-505) ID: max
24(507-508) ID: i1
24(509-509) COMMA: ,
24(511-512) ID: f2
24(513-513) RIGHT_PARENTHESIS: )
24(515-515) MINUS: -
24(517-518) ID: f2
24(519-519) PIPE: |
24(521-521) LESS_THAN: <
24(523-527) REAL: 0.001
24(528-528) NEWLINE: \n
25(529-533) ID: debug
25(535-535) PIPE: |
25(536-536) LEFT_PARENTHESIS: (
25(537-539) ID: max
25(541-542) ID: i2
25(543-543) COMMA: ,
25(545-546) ID: f2
25(547-547) RIGHT_PARENTHESIS: )
25(549-549) MINUS: -
25(551-552) ID: f2
25(553-553) PIPE: |
25(555-555) LESS_THAN: <
25(557-561) REAL: 0.001
25(562-562) NEWLINE: \n
26(563-563) END_OF_FILE: END OF FILE
