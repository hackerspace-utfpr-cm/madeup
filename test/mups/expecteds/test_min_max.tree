(block (define-variable f1 (REAL 8.6)) (define-variable f2 (REAL 9.1)) (define-variable i1 (INTEGER 3)) (define-variable i2 (INTEGER 4)) (call debug (< (abs (- (call min (call-with-names f1) (call-with-names f2)) (call-with-names f1))) (REAL 0.001))) (call debug (< (abs (- (call max (call-with-names f1) (call-with-names f2)) (call-with-names f2))) (REAL 0.001))) (call debug (== (call min (call-with-names i1) (call-with-names i2)) (call-with-names i1))) (call debug (== (call max (call-with-names i1) (call-with-names i2)) (call-with-names i2))) (call debug (< (abs (- (call min (call-with-names f1) (call-with-names i1)) (call-with-names i1))) (REAL 0.001))) (call debug (< (abs (- (call min (call-with-names f1) (call-with-names i2)) (call-with-names i2))) (REAL 0.001))) (call debug (< (abs (- (call max (call-with-names f1) (call-with-names i1)) (call-with-names f1))) (REAL 0.001))) (call debug (< (abs (- (call max (call-with-names f1) (call-with-names i2)) (call-with-names f1))) (REAL 0.001))) (call debug (< (abs (- (call min (call-with-names i1) (call-with-names f1)) (call-with-names i1))) (REAL 0.001))) (call debug (< (abs (- (call min (call-with-names i2) (call-with-names f1)) (call-with-names i2))) (REAL 0.001))) (call debug (< (abs (- (call max (call-with-names i1) (call-with-names f1)) (call-with-names f1))) (REAL 0.001))) (call debug (< (abs (- (call max (call-with-names i2) (call-with-names f1)) (call-with-names f1))) (REAL 0.001))) (call debug (< (abs (- (call min (call-with-names i1) (call-with-names f2)) (call-with-names i1))) (REAL 0.001))) (call debug (< (abs (- (call min (call-with-names i2) (call-with-names f2)) (call-with-names i2))) (REAL 0.001))) (call debug (< (abs (- (call max (call-with-names i1) (call-with-names f2)) (call-with-names f2))) (REAL 0.001))) (call debug (< (abs (- (call max (call-with-names i2) (call-with-names f2)) (call-with-names f2))) (REAL 0.001))))
