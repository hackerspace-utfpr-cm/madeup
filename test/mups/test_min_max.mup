f1 = 8.6
f2 = 9.1

i1 = 3
i2 = 4

debug |(min f1, f2) - f1| < 0.001
debug |(max f1, f2) - f2| < 0.001
debug (min i1, i2) == i1
debug (max i1, i2) == i2

debug |(min f1, i1) - i1| < 0.001
debug |(min f1, i2) - i2| < 0.001
debug |(max f1, i1) - f1| < 0.001
debug |(max f1, i2) - f1| < 0.001

debug |(min i1, f1) - i1| < 0.001
debug |(min i2, f1) - i2| < 0.001
debug |(max i1, f1) - f1| < 0.001
debug |(max i2, f1) - f1| < 0.001

debug |(min i1, f2) - i1| < 0.001
debug |(min i2, f2) - i2| < 0.001
debug |(max i1, f2) - f2| < 0.001
debug |(max i2, f2) - f2| < 0.001
