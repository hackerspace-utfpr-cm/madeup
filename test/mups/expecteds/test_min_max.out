|(min f1, f2) - f1| < 0.001: true
|(max f1, f2) - f2| < 0.001: true
(min i1, i2) == i1: true
(max i1, i2) == i2: true
|(min f1, i1) - i1| < 0.001: true
|(min f1, i2) - i2| < 0.001: true
|(max f1, i1) - f1| < 0.001: true
|(max f1, i2) - f1| < 0.001: true
|(min i1, f1) - i1| < 0.001: true
|(min i2, f1) - i2| < 0.001: true
|(max i1, f1) - f1| < 0.001: true
|(max i2, f1) - f1| < 0.001: true
|(min i1, f2) - i1| < 0.001: true
|(min i2, f2) - i2| < 0.001: true
|(max i1, f2) - f2| < 0.001: true
|(max i2, f2) - f2| < 0.001: true
